feat(search): sync with carbon react

IgnacioBecerra opened the pull request:
### Related Ticket(s)
[#10097](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10097)

### Description
This PR synchronizes our Search component with Carbon React's. 

### Changelog

**New**

- search can now be expandable
- search now accepts custom icon to replace magnifier

<!-- React and Web Component deploy previews are enabled by default. -->
<!-- To enable additional available deploy previews, apply the following -->
<!-- labels for the corresponding package: -->
<!-- *** "test: e2e": Codesandbox examples and e2e integration tests -->
<!-- *** "package: services": Services -->
<!-- *** "package: utilities": Utilities -->
<!-- *** "RTL": React / Web Components (RTL) -->
<!-- *** "feature flag": React / Web Components (experimental) -->

ariellalgilmore commented on packages/web-components/src/components/search/search.ts:
```diff
@@ -8,7 +8,7 @@
 
 import { customElement } from 'lit/decorators.js';
-import BXSearch from '../../internal/vendor/@carbon/web-components/components/search/search.js';
+import CDSSearch from '../../internal/vendor/@carbon/web-components/components/search/search.js';
```
i think we might want to keep it as `BXSearch` for now until we update `web-components` to v2 since it's still pulling from `@carbon/web-components: 1.xx`

IgnacioBecerra replied:
Completely missed this while I was working on the changes yesterday, but changed it back, good call!

oliviaflory commented:
@IgnacioBecerra looking good! 

A few things I noticed:

### 1. Disabled state
- The search icon isn't using the disabled color token
- The close button is still interactive, but it should also be disabled according to the React examples 

![Screenshot 2023-04-17 at 12 05 20 PM](https://user-images.githubusercontent.com/45692622/232573668-2ce59994-ca86-4453-b845-9b55da07e25e.png)

### 2. Expandable stories
The search field text is overlapping with the search icon in the Expandable and Expandable with layer stories

![Screenshot 2023-04-17 at 12 02 41 PM](https://user-images.githubusercontent.com/45692622/232574102-e4d9777d-bc19-4a72-b1c6-a768149107a4.png)

### 3. Storybook
- What is the `Name (name)` knob? I didn't see it in React and couldn't tell what it was controlling in WC

IgnacioBecerra commented:
@oliviaflory Thanks for the review, addressed the comments! 
As for the `name` knob, its property sets the `name` of the actual input element to be used within a form element. You can see this when logging in to a website and your browser tries to autofill the email input with your actual email, because its `name` is `e-mail`, same with other information such as `password` or `address`. 

Since the React storybook doesn't have it, I will remove it from our CWC Storybook as well.
